build_rules.gradle

```diff
@@ -1253,6 +1254,125 @@ ext.applyGrpcNature = {
 
 /*************************************************************************************************/
 
+ext.applyPortabilityNature = {
+  println "applyPortabilityNature with " + (it ? "$it" : "default configuration") + " for project $project.name"
+  applyJavaNature(enableFindbugs: false, enableErrorProne: false, shadowClosure: {
+    // guava uses the com.google.common and com.google.thirdparty package namespaces
+    relocate "com.google.common", "org.apache.beam.vendor.guava.v20.com.google.common"
+    relocate "com.google.thirdparty", "org.apache.beam.vendor.guava.v20.com.google.thirdparty"
+
+    relocate "com.google.protobuf", "org.apache.beam.vendor.protobuf.v3.com.google.protobuf"
+    relocate "com.google.gson", "org.apache.beam.vendor.gson.v2.com.google.gson"
+    relocate "io.grpc", "org.apache.beam.vendor.grpc.v1.io.grpc"
+    relocate "com.google.auth", "org.apache.beam.vendor.google_auth_library_credentials.v0_9_1.com.google.auth"
```

Comment:
Add a function for formatting relocation path from various parts ("org.apache.beam", library name, package, etc) ?

Reply:
I don't think a function will help much. `getJavaRelocatedPath` did make sure things were project specific, but in this case we are sharing the `vendor` namespace across packages.